Can't compile gnome-firmware and kde discover

**Describe the bug**

First reported in https://gitlab.gnome.org/World/gnome-firmware/-/issues/67, but maybe it's more appropriate here.

I included #7987 but I still get the same build errors:

```
       > [40/42] Compiling C object src/gnome-firmware.p/gfu-main.c.o
       > FAILED: src/gnome-firmware.p/gfu-main.c.o
       > gcc -Isrc/gnome-firmware.p -Isrc -I../src -I. -I.. -I/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/include -I/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/include/glib-2.0 -I/nix/store/fxnchbq09sq279ssqdsgzigsyvp6hz06-glib-2.80.4/lib/glib-2.0/include -I/nix/store/4r3fsd2r23ags1gdpwz2dsrws6jy63m1-gdk-pixbuf-2.42.12-dev/include/gdk-pixbuf-2.0 -I/nix/store/r577cvxqgb2acrcv9vhw03293mwgvrq9-gtk4-4.14.5-dev/include/gtk-4.0 -I/nix/store/s9gm9vyza9yvfsz9ag01s4hw018gjhqg-cairo-1.18.2-dev/include/cairo -I/nix/store/hl14fvf6jpfv30j4p8hqgzqpmrmx2m7b-freetype-2.13.3-dev/include/freetype2 -I/nix/store/hl14fvf6jpfv30j4p8hqgzqpmrmx2m7b-freetype-2.13.3-dev/include -I/nix/store/nr8688frwk2jv3c9qwiz0ywi1979li62-graphene-1.10.8-dev/include/graphene-1.0 -I/nix/store/yvs7cfic7w4d67fpwbhgn6sxqfpzb4a1-graphene-1.10.8/lib/graphene-1.0/include -I/nix/store/na5ax7xn886i8mh91bzzda8hma313hwh-pango-1.52.2-dev/include/pango-1.0 -I/nix/store/z0g6vlvsrnsmbfq4bsp0pa4j63a69jcv-harfbuzz-9.0.0-dev/include/harfbuzz -I/nix/store/jrr54zy7h7w1lhh3rk5q4zcbfwz57r3c-vulkan-headers-1.3.290.0/include -I/nix/store/9768n9gki54vzs62p802kgjph89sxs05-fwupd-2.0.1-dev/include/fwupd-3 -I/nix/store/n5hh60n4kfkz2fzxdfgls9smzra7ldny-libxmlb-0.3.19-dev/include/libxmlb-2 -I/nix/store/5gcgm6z9k9jx2rf7r1bwngf73ma1xvcn-libadwaita-1.5.3-dev/include/libadwaita-1 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Waggregate-return -Warray-bounds -Wcast-align -Wclobbered -Wdeclaration-after-statement -Wempty-body -Wextra -Wformat=2 -Wformat-security -Wformat-signedness -Wignored-qualifiers -Wimplicit-function-declaration -Winit-self -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-discarded-qualifiers -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-suggest-attribute=format -Wno-unused-parameter -Wno-cast-function-type -Wold-style-definition -Wno-deprecated-declarations -Wno-format-nonliteral -Woverride-init -Wpacked -Wpointer-arith -Wredundant-decls -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wstrict-prototypes -Wswitch-default -Wtype-limits -Wundef -Wuninitialized -Wunused-but-set-variable -Wwrite-strings -mfpmath=sse -msse -msse2 -mfpmath=sse -msse -msse2 '-DG_LOG_DOMAIN="Gfu"' -MD -MQ src/gnome-firmware.p/gfu-main.c.o -MF src/gnome-firmware.p/gfu-main.c.o.d -o src/gnome-firmware.p/gfu-main.c.o -c ../src/gfu-main.c
       > In file included from /nix/store/9768n9gki54vzs62p802kgjph89sxs05-fwupd-2.0.1-dev/include/fwupd-3/fwupd.h:15,
       >                  from ../src/gfu-common.h:11,
       >                  from ../src/gfu-main.c:16:
       > /nix/store/9768n9gki54vzs62p802kgjph89sxs05-fwupd-2.0.1-dev/include/fwupd-3/libfwupd/fwupd-codec.h:10:10: fatal error: json-glib/json-glib.h: No such file or directory
       >    10 | #include <json-glib/json-glib.h>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > ninja: build stopped: subcommand failed.
```

**Additional questions**

- Operating system and version: NixOS
- Have you tried rebooting? N/A
- Is this a regression? Yes

@superm1

This is the full build log for gnome-firmware. There's some nix-specific output, but you can see the normal meson output as well.
It detects all the dependencies that are specified in the gnome-firmware meson build, and finds them in the environment, but there's no mention of json-glib.

```
Running phase: unpackPhase
@nix {"action":"setPhase","phase":"unpackPhase"}
unpacking source archive /nix/store/hhrh1k96x39i3psbardllz33qqh4jvws-source
source root is source
Running phase: patchPhase
@nix {"action":"setPhase","phase":"patchPhase"}
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix {"action":"setPhase","phase":"updateAutotoolsGnuConfigScriptsPhase"}
Running phase: configurePhase
@nix {"action":"setPhase","phase":"configurePhase"}
mesonConfigurePhase flags: --prefix=/nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0 --libdir=/nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/lib --libexecdir=/nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/libexec --bindir=/nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/bin --sbindir=/nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/sbin --includedir=/nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/include --mandir=/nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/share/man --infodir=/nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/share/info --localedir=/nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/share/locale -Dauto_features=enabled -Dwrap_mode=nodownload --buildtype=plain -Dconsolekit=false
The Meson build system
Version: 1.5.2
Source dir: /build/source
Build dir: /build/source/build
Build type: native build
Project name: gnome-firmware
Project version: 46.0
C compiler for the host machine: gcc (gcc 13.3.0 "gcc (GCC) 13.3.0")
C linker for the host machine: gcc ld.bfd 2.43.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C supports arguments -Waggregate-return: YES 
Compiler for C supports arguments -Warray-bounds: YES 
Compiler for C supports arguments -Wcast-align: YES 
Compiler for C supports arguments -Wclobbered: YES 
Compiler for C supports arguments -Wdeclaration-after-statement: YES 
Compiler for C supports arguments -Wempty-body: YES 
Compiler for C supports arguments -Wextra: YES 
meson.build:60: WARNING: Consider using the built-in warning_level option instead of using "-Wextra".
Compiler for C supports arguments -Wformat=2: YES 
Compiler for C supports arguments -Wformat-security: YES 
Compiler for C supports arguments -Wformat-signedness: YES 
Compiler for C supports arguments -Wignored-qualifiers: YES 
Compiler for C supports arguments -Wimplicit-function-declaration: YES 
Compiler for C supports arguments -Winit-self: YES 
Compiler for C supports arguments -Wmissing-declarations: YES 
Compiler for C supports arguments -Wmissing-format-attribute: YES 
Compiler for C supports arguments -Wmissing-include-dirs: YES 
Compiler for C supports arguments -Wmissing-noreturn: YES 
Compiler for C supports arguments -Wmissing-parameter-type: YES 
Compiler for C supports arguments -Wmissing-prototypes: YES 
Compiler for C supports arguments -Wnested-externs: YES 
Compiler for C supports arguments -Wno-discarded-qualifiers: YES 
Compiler for C supports arguments -Wno-missing-field-initializers: YES 
Compiler for C supports arguments -Wno-strict-aliasing: YES 
Compiler for C supports arguments -Wno-suggest-attribute=format: YES 
Compiler for C supports arguments -Wno-unused-parameter: YES 
Compiler for C supports arguments -Wno-cast-function-type: YES 
Compiler for C supports arguments -Wold-style-definition: YES 
Compiler for C supports arguments -Wno-deprecated-declarations: YES 
Compiler for C supports arguments -Wno-format-nonliteral: YES 
Compiler for C supports arguments -Woverride-init: YES 
Compiler for C supports arguments -Wpacked: YES 
Compiler for C supports arguments -Wpointer-arith: YES 
Compiler for C supports arguments -Wredundant-decls: YES 
Compiler for C supports arguments -Wreturn-type: YES 
Compiler for C supports arguments -Wshadow: YES 
Compiler for C supports arguments -Wsign-compare: YES 
Compiler for C supports arguments -Wstrict-aliasing: YES 
Compiler for C supports arguments -Wstrict-prototypes: YES 
Compiler for C supports arguments -Wswitch-default: YES 
Compiler for C supports arguments -Wtype-limits: YES 
Compiler for C supports arguments -Wundef: YES 
Compiler for C supports arguments -Wuninitialized: YES 
Compiler for C supports arguments -Wunused-but-set-variable: YES 
Compiler for C supports arguments -Wwrite-strings: YES 
Compiler for C supports link arguments -Wl,-z,relro: YES 
Compiler for C supports link arguments -Wl,-z,now: YES 
Found pkg-config: YES (/nix/store/mi7q9m86m3rkqmc0niyvfv25ji9fxxha-pkg-config-wrapper-0.29.2/bin/pkg-config) 0.29.2
Run-time dependency gtk4 found: YES 4.14.5
Run-time dependency gio-2.0 found: YES 2.80.4
Run-time dependency fwupd found: YES 2.0.1
Run-time dependency xmlb found: YES 0.3.19
Run-time dependency libadwaita-1 found: YES 1.5.3
Configuring config.h using configuration
Dependency gio-2.0 found: YES 2.80.4 (cached)
Program /nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/bin/glib-compile-resources found: YES (/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/bin/glib-compile-resources)
Program help2man found: YES (/nix/store/db214rydhjz1z9g0g3hq41x40xxb7jni-help2man-1.49.3/bin/help2man)
Program msgfmt found: YES (/nix/store/7dkf6h889q566b5zh6z2bz13916rp7qf-gettext-0.21.1/bin/msgfmt)
Program msginit found: YES (/nix/store/7dkf6h889q566b5zh6z2bz13916rp7qf-gettext-0.21.1/bin/msginit)
Program msgmerge found: YES (/nix/store/7dkf6h889q566b5zh6z2bz13916rp7qf-gettext-0.21.1/bin/msgmerge)
Program xgettext found: YES (/nix/store/7dkf6h889q566b5zh6z2bz13916rp7qf-gettext-0.21.1/bin/xgettext)
Program appstreamcli found: NO
Build targets in project: 40
NOTICE: Future-deprecated features used:
 * 0.56.0: {'meson.source_root'}

gnome-firmware 46.0

  User defined options
    auto_features: enabled
    bindir       : /nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/bin
    buildtype    : plain
    includedir   : /nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/include
    infodir      : /nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/share/info
    libdir       : /nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/lib
    libexecdir   : /nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/libexec
    localedir    : /nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/share/locale
    mandir       : /nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/share/man
    prefix       : /nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0
    sbindir      : /nix/store/jd3lcn0hpfbr5ihhv7avkx7k5fif0pmm-gnome-firmware-46.0/sbin
    wrap_mode    : nodownload
    consolekit   : false

Found ninja-1.12.1 at /nix/store/xhx4mn5hmwjw95s2yqqjlbbrb3p1zdlp-ninja-1.12.1/bin/ninja
mesonConfigurePhase: enabled\ parallel\ building
Running phase: buildPhase
@nix {"action":"setPhase","phase":"buildPhase"}
build flags: -j48
[1/42] Building translation po/ca/LC_MESSAGES/gnome-firmware-ca.mo
[2/42] Building translation po/da/LC_MESSAGES/gnome-firmware-da.mo
[3/42] Building translation po/de/LC_MESSAGES/gnome-firmware-de.mo
[4/42] Building translation po/en_GB/LC_MESSAGES/gnome-firmware-en_GB.mo
[5/42] Building translation po/cs/LC_MESSAGES/gnome-firmware-cs.mo
[6/42] Building translation po/es/LC_MESSAGES/gnome-firmware-es.mo
[7/42] Building translation po/eu/LC_MESSAGES/gnome-firmware-eu.mo
[8/42] Building translation po/fa/LC_MESSAGES/gnome-firmware-fa.mo
[9/42] Building translation po/fi/LC_MESSAGES/gnome-firmware-fi.mo
[10/42] Building translation po/fr/LC_MESSAGES/gnome-firmware-fr.mo
[11/42] Building translation po/fur/LC_MESSAGES/gnome-firmware-fur.mo
[12/42] Building translation po/gl/LC_MESSAGES/gnome-firmware-gl.mo
[13/42] Building translation po/hu/LC_MESSAGES/gnome-firmware-hu.mo
[14/42] Generating src/gnome-firmware-resources_c with a custom command
xml-stripblanks preprocessing requested, but XMLLINT is not set, and xmllint is not in PATH
[15/42] Generating src/gnome-firmware-resources_h with a custom command
xml-stripblanks preprocessing requested, but XMLLINT is not set, and xmllint is not in PATH
[16/42] Building translation po/he/LC_MESSAGES/gnome-firmware-he.mo
[17/42] Building translation po/it/LC_MESSAGES/gnome-firmware-it.mo
[18/42] Building translation po/hr/LC_MESSAGES/gnome-firmware-hr.mo
[19/42] Building translation po/id/LC_MESSAGES/gnome-firmware-id.mo
[20/42] Building translation po/pa/LC_MESSAGES/gnome-firmware-pa.mo
[21/42] Building translation po/pl/LC_MESSAGES/gnome-firmware-pl.mo
[22/42] Building translation po/nl/LC_MESSAGES/gnome-firmware-nl.mo
[23/42] Building translation po/ka/LC_MESSAGES/gnome-firmware-ka.mo
[24/42] Building translation po/pt_BR/LC_MESSAGES/gnome-firmware-pt_BR.mo
[25/42] Building translation po/ru/LC_MESSAGES/gnome-firmware-ru.mo
[26/42] Building translation po/sr/LC_MESSAGES/gnome-firmware-sr.mo
[27/42] Building translation po/sv/LC_MESSAGES/gnome-firmware-sv.mo
[28/42] Building translation po/tr/LC_MESSAGES/gnome-firmware-tr.mo
[29/42] Building translation po/sk/LC_MESSAGES/gnome-firmware-sk.mo
[30/42] Building translation po/sl/LC_MESSAGES/gnome-firmware-sl.mo
[31/42] Building translation po/zh_CN/LC_MESSAGES/gnome-firmware-zh_CN.mo
[32/42] Building translation po/uk/LC_MESSAGES/gnome-firmware-uk.mo
[33/42] Building translation po/zh_TW/LC_MESSAGES/gnome-firmware-zh_TW.mo
[34/42] Merging translations for data/org.gnome.Firmware.desktop
[35/42] Merging translations for data/appdata/org.gnome.Firmware.metainfo.xml
[36/42] Compiling C object src/gnome-firmware.p/meson-generated_.._gnome-firmware-resources.c.o
[37/42] Compiling C object src/gnome-firmware.p/gfu-device-row.c.o
FAILED: src/gnome-firmware.p/gfu-device-row.c.o 
gcc -Isrc/gnome-firmware.p -Isrc -I../src -I. -I.. -I/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/include -I/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/include/glib-2.0 -I/nix/store/fxnchbq09sq279ssqdsgzigsyvp6hz06-glib-2.80.4/lib/glib-2.0/include -I/nix/store/4r3fsd2r23ags1gdpwz2dsrws6jy63m1-gdk-pixbuf-2.42.12-dev/include/gdk-pixbuf-2.0 -I/nix/store/r577cvxqgb2acrcv9vhw03293mwgvrq9-gtk4-4.14.5-dev/include/gtk-4.0 -I/nix/store/s9gm9vyza9yvfsz9ag01s4hw018gjhqg-cairo-1.18.2-dev/include/cairo -I/nix/store/hl14fvf6jpfv30j4p8hqgzqpmrmx2m7b-freetype-2.13.3-dev/include/freetype2 -I/nix/store/hl14fvf6jpfv30j4p8hqgzqpmrmx2m7b-freetype-2.13.3-dev/include -I/nix/store/nr8688frwk2jv3c9qwiz0ywi1979li62-graphene-1.10.8-dev/include/graphene-1.0 -I/nix/store/yvs7cfic7w4d67fpwbhgn6sxqfpzb4a1-graphene-1.10.8/lib/graphene-1.0/include -I/nix/store/na5ax7xn886i8mh91bzzda8hma313hwh-pango-1.52.2-dev/include/pango-1.0 -I/nix/store/z0g6vlvsrnsmbfq4bsp0pa4j63a69jcv-harfbuzz-9.0.0-dev/include/harfbuzz -I/nix/store/jrr54zy7h7w1lhh3rk5q4zcbfwz57r3c-vulkan-headers-1.3.290.0/include -I/nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3 -I/nix/store/n5hh60n4kfkz2fzxdfgls9smzra7ldny-libxmlb-0.3.19-dev/include/libxmlb-2 -I/nix/store/5gcgm6z9k9jx2rf7r1bwngf73ma1xvcn-libadwaita-1.5.3-dev/include/libadwaita-1 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Waggregate-return -Warray-bounds -Wcast-align -Wclobbered -Wdeclaration-after-statement -Wempty-body -Wextra -Wformat=2 -Wformat-security -Wformat-signedness -Wignored-qualifiers -Wimplicit-function-declaration -Winit-self -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-discarded-qualifiers -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-suggest-attribute=format -Wno-unused-parameter -Wno-cast-function-type -Wold-style-definition -Wno-deprecated-declarations -Wno-format-nonliteral -Woverride-init -Wpacked -Wpointer-arith -Wredundant-decls -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wstrict-prototypes -Wswitch-default -Wtype-limits -Wundef -Wuninitialized -Wunused-but-set-variable -Wwrite-strings -mfpmath=sse -msse -msse2 -mfpmath=sse -msse -msse2 '-DG_LOG_DOMAIN="Gfu"' -MD -MQ src/gnome-firmware.p/gfu-device-row.c.o -MF src/gnome-firmware.p/gfu-device-row.c.o.d -o src/gnome-firmware.p/gfu-device-row.c.o -c ../src/gfu-device-row.c
In file included from /nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3/fwupd.h:15,
                 from ../src/gfu-device-row.h:9,
                 from ../src/gfu-device-row.c:9:
/nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3/libfwupd/fwupd-codec.h:10:10: fatal error: json-glib/json-glib.h: No such file or directory
   10 | #include <json-glib/json-glib.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[38/42] Compiling C object src/gnome-firmware.p/gfu-release-row.c.o
FAILED: src/gnome-firmware.p/gfu-release-row.c.o 
gcc -Isrc/gnome-firmware.p -Isrc -I../src -I. -I.. -I/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/include -I/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/include/glib-2.0 -I/nix/store/fxnchbq09sq279ssqdsgzigsyvp6hz06-glib-2.80.4/lib/glib-2.0/include -I/nix/store/4r3fsd2r23ags1gdpwz2dsrws6jy63m1-gdk-pixbuf-2.42.12-dev/include/gdk-pixbuf-2.0 -I/nix/store/r577cvxqgb2acrcv9vhw03293mwgvrq9-gtk4-4.14.5-dev/include/gtk-4.0 -I/nix/store/s9gm9vyza9yvfsz9ag01s4hw018gjhqg-cairo-1.18.2-dev/include/cairo -I/nix/store/hl14fvf6jpfv30j4p8hqgzqpmrmx2m7b-freetype-2.13.3-dev/include/freetype2 -I/nix/store/hl14fvf6jpfv30j4p8hqgzqpmrmx2m7b-freetype-2.13.3-dev/include -I/nix/store/nr8688frwk2jv3c9qwiz0ywi1979li62-graphene-1.10.8-dev/include/graphene-1.0 -I/nix/store/yvs7cfic7w4d67fpwbhgn6sxqfpzb4a1-graphene-1.10.8/lib/graphene-1.0/include -I/nix/store/na5ax7xn886i8mh91bzzda8hma313hwh-pango-1.52.2-dev/include/pango-1.0 -I/nix/store/z0g6vlvsrnsmbfq4bsp0pa4j63a69jcv-harfbuzz-9.0.0-dev/include/harfbuzz -I/nix/store/jrr54zy7h7w1lhh3rk5q4zcbfwz57r3c-vulkan-headers-1.3.290.0/include -I/nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3 -I/nix/store/n5hh60n4kfkz2fzxdfgls9smzra7ldny-libxmlb-0.3.19-dev/include/libxmlb-2 -I/nix/store/5gcgm6z9k9jx2rf7r1bwngf73ma1xvcn-libadwaita-1.5.3-dev/include/libadwaita-1 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Waggregate-return -Warray-bounds -Wcast-align -Wclobbered -Wdeclaration-after-statement -Wempty-body -Wextra -Wformat=2 -Wformat-security -Wformat-signedness -Wignored-qualifiers -Wimplicit-function-declaration -Winit-self -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-discarded-qualifiers -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-suggest-attribute=format -Wno-unused-parameter -Wno-cast-function-type -Wold-style-definition -Wno-deprecated-declarations -Wno-format-nonliteral -Woverride-init -Wpacked -Wpointer-arith -Wredundant-decls -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wstrict-prototypes -Wswitch-default -Wtype-limits -Wundef -Wuninitialized -Wunused-but-set-variable -Wwrite-strings -mfpmath=sse -msse -msse2 -mfpmath=sse -msse -msse2 '-DG_LOG_DOMAIN="Gfu"' -MD -MQ src/gnome-firmware.p/gfu-release-row.c.o -MF src/gnome-firmware.p/gfu-release-row.c.o.d -o src/gnome-firmware.p/gfu-release-row.c.o -c ../src/gfu-release-row.c
In file included from /nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3/fwupd.h:15,
                 from ../src/gfu-release-row.h:9,
                 from ../src/gfu-release-row.c:11:
/nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3/libfwupd/fwupd-codec.h:10:10: fatal error: json-glib/json-glib.h: No such file or directory
   10 | #include <json-glib/json-glib.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[39/42] Compiling C object src/gnome-firmware.p/gfu-common.c.o
FAILED: src/gnome-firmware.p/gfu-common.c.o 
gcc -Isrc/gnome-firmware.p -Isrc -I../src -I. -I.. -I/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/include -I/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/include/glib-2.0 -I/nix/store/fxnchbq09sq279ssqdsgzigsyvp6hz06-glib-2.80.4/lib/glib-2.0/include -I/nix/store/4r3fsd2r23ags1gdpwz2dsrws6jy63m1-gdk-pixbuf-2.42.12-dev/include/gdk-pixbuf-2.0 -I/nix/store/r577cvxqgb2acrcv9vhw03293mwgvrq9-gtk4-4.14.5-dev/include/gtk-4.0 -I/nix/store/s9gm9vyza9yvfsz9ag01s4hw018gjhqg-cairo-1.18.2-dev/include/cairo -I/nix/store/hl14fvf6jpfv30j4p8hqgzqpmrmx2m7b-freetype-2.13.3-dev/include/freetype2 -I/nix/store/hl14fvf6jpfv30j4p8hqgzqpmrmx2m7b-freetype-2.13.3-dev/include -I/nix/store/nr8688frwk2jv3c9qwiz0ywi1979li62-graphene-1.10.8-dev/include/graphene-1.0 -I/nix/store/yvs7cfic7w4d67fpwbhgn6sxqfpzb4a1-graphene-1.10.8/lib/graphene-1.0/include -I/nix/store/na5ax7xn886i8mh91bzzda8hma313hwh-pango-1.52.2-dev/include/pango-1.0 -I/nix/store/z0g6vlvsrnsmbfq4bsp0pa4j63a69jcv-harfbuzz-9.0.0-dev/include/harfbuzz -I/nix/store/jrr54zy7h7w1lhh3rk5q4zcbfwz57r3c-vulkan-headers-1.3.290.0/include -I/nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3 -I/nix/store/n5hh60n4kfkz2fzxdfgls9smzra7ldny-libxmlb-0.3.19-dev/include/libxmlb-2 -I/nix/store/5gcgm6z9k9jx2rf7r1bwngf73ma1xvcn-libadwaita-1.5.3-dev/include/libadwaita-1 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Waggregate-return -Warray-bounds -Wcast-align -Wclobbered -Wdeclaration-after-statement -Wempty-body -Wextra -Wformat=2 -Wformat-security -Wformat-signedness -Wignored-qualifiers -Wimplicit-function-declaration -Winit-self -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-discarded-qualifiers -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-suggest-attribute=format -Wno-unused-parameter -Wno-cast-function-type -Wold-style-definition -Wno-deprecated-declarations -Wno-format-nonliteral -Woverride-init -Wpacked -Wpointer-arith -Wredundant-decls -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wstrict-prototypes -Wswitch-default -Wtype-limits -Wundef -Wuninitialized -Wunused-but-set-variable -Wwrite-strings -mfpmath=sse -msse -msse2 -mfpmath=sse -msse -msse2 '-DG_LOG_DOMAIN="Gfu"' -MD -MQ src/gnome-firmware.p/gfu-common.c.o -MF src/gnome-firmware.p/gfu-common.c.o.d -o src/gnome-firmware.p/gfu-common.c.o -c ../src/gfu-common.c
In file included from /nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3/fwupd.h:15,
                 from ../src/gfu-common.h:11,
                 from ../src/gfu-common.c:12:
/nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3/libfwupd/fwupd-codec.h:10:10: fatal error: json-glib/json-glib.h: No such file or directory
   10 | #include <json-glib/json-glib.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[40/42] Compiling C object src/gnome-firmware.p/gfu-main.c.o
FAILED: src/gnome-firmware.p/gfu-main.c.o 
gcc -Isrc/gnome-firmware.p -Isrc -I../src -I. -I.. -I/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/include -I/nix/store/82bbzzkj4ncqg1q08f9c6f91pam9kf0n-glib-2.80.4-dev/include/glib-2.0 -I/nix/store/fxnchbq09sq279ssqdsgzigsyvp6hz06-glib-2.80.4/lib/glib-2.0/include -I/nix/store/4r3fsd2r23ags1gdpwz2dsrws6jy63m1-gdk-pixbuf-2.42.12-dev/include/gdk-pixbuf-2.0 -I/nix/store/r577cvxqgb2acrcv9vhw03293mwgvrq9-gtk4-4.14.5-dev/include/gtk-4.0 -I/nix/store/s9gm9vyza9yvfsz9ag01s4hw018gjhqg-cairo-1.18.2-dev/include/cairo -I/nix/store/hl14fvf6jpfv30j4p8hqgzqpmrmx2m7b-freetype-2.13.3-dev/include/freetype2 -I/nix/store/hl14fvf6jpfv30j4p8hqgzqpmrmx2m7b-freetype-2.13.3-dev/include -I/nix/store/nr8688frwk2jv3c9qwiz0ywi1979li62-graphene-1.10.8-dev/include/graphene-1.0 -I/nix/store/yvs7cfic7w4d67fpwbhgn6sxqfpzb4a1-graphene-1.10.8/lib/graphene-1.0/include -I/nix/store/na5ax7xn886i8mh91bzzda8hma313hwh-pango-1.52.2-dev/include/pango-1.0 -I/nix/store/z0g6vlvsrnsmbfq4bsp0pa4j63a69jcv-harfbuzz-9.0.0-dev/include/harfbuzz -I/nix/store/jrr54zy7h7w1lhh3rk5q4zcbfwz57r3c-vulkan-headers-1.3.290.0/include -I/nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3 -I/nix/store/n5hh60n4kfkz2fzxdfgls9smzra7ldny-libxmlb-0.3.19-dev/include/libxmlb-2 -I/nix/store/5gcgm6z9k9jx2rf7r1bwngf73ma1xvcn-libadwaita-1.5.3-dev/include/libadwaita-1 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Waggregate-return -Warray-bounds -Wcast-align -Wclobbered -Wdeclaration-after-statement -Wempty-body -Wextra -Wformat=2 -Wformat-security -Wformat-signedness -Wignored-qualifiers -Wimplicit-function-declaration -Winit-self -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-discarded-qualifiers -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-suggest-attribute=format -Wno-unused-parameter -Wno-cast-function-type -Wold-style-definition -Wno-deprecated-declarations -Wno-format-nonliteral -Woverride-init -Wpacked -Wpointer-arith -Wredundant-decls -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wstrict-prototypes -Wswitch-default -Wtype-limits -Wundef -Wuninitialized -Wunused-but-set-variable -Wwrite-strings -mfpmath=sse -msse -msse2 -mfpmath=sse -msse -msse2 '-DG_LOG_DOMAIN="Gfu"' -MD -MQ src/gnome-firmware.p/gfu-main.c.o -MF src/gnome-firmware.p/gfu-main.c.o.d -o src/gnome-firmware.p/gfu-main.c.o -c ../src/gfu-main.c
In file included from /nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3/fwupd.h:15,
                 from ../src/gfu-common.h:11,
                 from ../src/gfu-main.c:16:
/nix/store/xns5kkmislmm5vyrgqzxzndidl30bygk-fwupd-2.0.1-dev/include/fwupd-3/libfwupd/fwupd-codec.h:10:10: fatal error: json-glib/json-glib.h: No such file or directory
   10 | #include <json-glib/json-glib.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```

--- a/libfwupd/meson.build
+++ b/libfwupd/meson.build
@@ -88,7 +88,7 @@ libfwupd_dep = declare_dependency(
 pkgg = import('pkgconfig')
 pkgg.generate(
   fwupd,
-  requires: [ 'gio-2.0' ],
+  requires: [ 'gio-2.0', 'json-glib-1.0' ],
   subdirs: base_dir,
   version: meson.project_version(),
   name: 'fwupd',

